### PR TITLE
Fix: default logger and providers

### PR DESCRIPTION
**Pull request by JoeHowarth**
Problems encountered when running GR in tilt 
- mergeDeep handles cycles

### CODE REVIEW

File: relayer/utils.ts [hunk 1]
```diff
@@ -57,8 +57,8 @@ export function parseVaaWithBytes(bytes: SignedVaa): ParsedVaaWithBytes {
  * @param target
  * @param ...sources
  */
-export function mergeDeep<T>(target: Partial<T>, ...sources: Partial<T>[]): T {
-  if (!sources.length) {
+export function mergeDeep<T>(target: Partial<T>, sources: Partial<T>[], maxDepth = 10): T {
```
Comment (JoeHowarth) [inline relayer/utils.ts:60]: Without max depth this fails when objects have cycles

File: relayer/storage.ts [hunk 2]
```diff
@@ -97,6 +97,11 @@ export class Storage<T extends Context> {
 
   constructor(private relayer: RelayerApp<T>, opts: StorageOptions) {
     this.opts = Object.assign({}, defaultOptions, opts);
+    // ensure redis is defined 
+    if (!this.opts.redis) {
```
Comment (JoeHowarth) [inline relayer/storage.ts:101]: opts.redis can be undefined even though the defaults have it defined as `redis: {}` due to how Object.assign handles {redis: undefined} vs {}

File: relayer/application-standard.ts [hunk 3]
```diff
@@ -40,7 +41,8 @@ const defaultOpts: Partial<StandardRelayerAppOpts> = {
   workflows: {
     retries: 3,
   },
-  fetchSourceTxhash: true
+  fetchSourceTxhash: true,
+  logger: defaultLogger,
```
Comment (JoeHowarth) [inline relayer/application-standard.ts:45]: If we don't have this we get an error if the opt is not specified

File: relayer/index.ts [hunk 4]
```diff
@@ -3,4 +3,5 @@ export * from "./application-standard";
 export * from "./context";
 export * from "./compose.middleware";
 export * from "./storage";
+export * from './logging'
```
Comment (JoeHowarth) [inline relayer/index.ts:6]: export the default logger

File: relayer/application-standard.ts (hunk 3 above)
Comment (solanoepalacio) [reply relayer/application-standard.ts:45]: should we log anything if no logger is passed?

Comment (JoeHowarth) [reply relayer/application-standard.ts:45]: Yeah actually I think there are 2 things here:
- by default we should still log in a sane way
- the user should be able to turn logging off entirely and not cause an exception 

So imo we should still have the above code, but additionally track down the source of the exceptions. Thoughts?

Comment (solanoepalacio) [reply relayer/application-standard.ts:45]: this is already merged, but for whatever is worth:
I think that this module shouldn't log anything by default. In the case we receive a logger, then use that logger to log, which is a way of saying "pass a logger to turn on the logs".
If we receive no logger, then we instantiate a logger mock that doesn't log, and then we don't have to worry about the exceptions. Is this pattern to ugly?

**Pull request merged**